pkg/api/login_handler.go

```diff
@@ -21,12 +21,32 @@ const (
 	JWTCookieName          = "access_token"
 )
 
-func GenerateJWT(secret []byte, userID int, issuedAt, expiresAt time.Time) (string, error) {
+func generateJWT(claims *jwt.StandardClaims, secret []byte) (string, error) {
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	return token.SignedString(secret)
+}
+
+// GenerateJWTLogin creates a jwt token which can be used for authentication during login only, i.e. it will not work for password reset.
+// It supports backward compatibility for creating a login jwt. The audience is set to LoginAudience which is
+// an empty string and no email is passed to support the ability of login for users via user/access keys which don't have an email set
+func GenerateJWTLogin(secret []byte, userID int, issuedAt, expiresAt time.Time) (string, error) {
 	claims := &jwt.StandardClaims{
+		Audience:  LoginAudience,
 		Subject:   fmt.Sprint(userID),
 		IssuedAt:  issuedAt.Unix(),
 		ExpiresAt: expiresAt.Unix(),
 	}
-	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
-	return token.SignedString(secret)
+	return generateJWT(claims, secret)
+}
+
+// GenerateJWTResetPassword creates a jwt token with the field subject set the email passed.
+func GenerateJWTResetPassword(secret []byte, aud string, userID int, email string, issuedAt, expiresAt time.Time) (string, error) {
+	claims := &jwt.StandardClaims{
+		Audience:  aud,
```

Comment:
The audience should be set by the implementation as the func name, we need to set the audience because this is not the generic way.

Reply:
ok, fixed